File: Primes.java
class Primes {
    public static void main(String[] args){

        System.out.print("\033[H\033[2J");


        final double startTime = System.currentTimeMillis();

        int numOfPrimes = 10000;
        int numsPrinted = 0;
        int numberIterator = 2;
        int recordedPrimes[] = new int[numOfPrimes+1];
        int ceilingCheck;
        
        boolean composite = false;
        while (numsPrinted<numOfPrimes){
            ceilingCheck = numberIterator;
            composite = false;

            for (int i = 0; (recordedPrimes[i]<ceilingCheck)&(recordedPrimes[i]!=0); i++){
                if (numberIterator%recordedPrimes[i]==0){
                    composite = true;
                }
                else {
                    ceilingCheck=numberIterator/recordedPrimes[i];
                }
            }

            if (!composite){
                System.out.println(numberIterator);
                recordedPrimes[numsPrinted]=numberIterator;
                numsPrinted++;
            }
                
            numberIterator++;

        }

        final double endTime = System.currentTimeMillis();

        System.out.println("--- " + ((endTime - startTime)/1000) + " seconds ---");
    }
}

File: Primes.py
import numpy as np

import time
start_time = time.time()
numOfPrimes = 100000
numsPrinted = 0
numberIterator = 2
recordedPrimes = np.zeros(numOfPrimes+1)

composite = False
while numsPrinted<numOfPrimes:
    celingCheck = numberIterator**0.5
    composite = False

    i = 0
    while recordedPrimes[i]<=celingCheck and recordedPrimes[i]!=0:
        if numberIterator%recordedPrimes[i]==0:
            composite = True
        i+=1
    if not composite:
        print(numberIterator)
        recordedPrimes[numsPrinted]=numberIterator
        numsPrinted+=1
    numberIterator+=1
end_time = time.time()
print("--- %s seconds ---" % (end_time-start_time))